packages/valory/skills/check_stop_trading_abci/tests/test_rounds.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+# ------------------------------------------------------------------------------
+#
+#   Copyright 2024 Valory AG
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This module contains the tests for the check stop trading ABCI application."""
+
+import pytest
+import logging  # noqa: F401
+from unittest.mock import MagicMock
+from typing import Set
+from packages.valory.skills.abstract_round_abci.base import (
+    AbciApp,
+    AbciAppTransitionFunction,
+    AbstractRound,
+    AppState,
+    BaseSynchronizedData,
+    CollectionRound,
+    DegenerateRound,
+    DeserializedCollection,
+    VotingRound,
+    get_name,
+)
+
+from packages.valory.skills.check_stop_trading_abci.rounds import (
+    CheckStopTradingRound,
+    Event,
+    FinishedCheckStopTradingRound,
+    FinishedWithSkipTradingRound,
+    SynchronizedData,
+    CheckStopTradingAbciApp
+)
+ 
+
+from packages.valory.skills.check_stop_trading_abci.payloads import (
+    CheckStopTradingPayload,
+)
+
+@pytest.fixture
+def synchronized_data():
+    """Get the synchronized data"""
+    return SynchronizedData(db=dict())
+
+
+@pytest.fixture
+def abci_app():
+    """Fixture to get the ABCI app with necessary parameters."""
+    # Create mocks for the required parameters
+    synchronized_data = MagicMock()
+    logger = MagicMock()
+    context = MagicMock()
+
+    # Instantiate CheckStopTradingAbciApp with mocked parameters
+    return CheckStopTradingAbciApp(synchronized_data=synchronized_data, logger=logger, context=context)
+
+
+
+def test_check_stop_trading_round_initialization(synchronized_data):
```

Comment:
These tests should also check whether we can run the rounds and that for every potential execution branch  we get the expected data and event. Check this for some inspiration: https://github.com/valory-xyz/IEKit/blob/main/packages/valory/skills/dynamic_nft_abci/tests/test_rounds.py#L125